src/dawn/CodeGen/CodeGen.cpp

```diff
@@ -27,6 +28,208 @@ size_t CodeGen::getVerticalTmpHaloSizeForMultipleStencils(
              : 0;
 }
 
+std::string CodeGen::generateGlobals(std::shared_ptr<SIR> const& sir,
+                                     std::string namespace_) const {
+
+  const auto& globalsMap = *(sir->GlobalVariableMap);
+  if(globalsMap.empty())
+    return "";
+
+  std::stringstream ss;
+
+  Namespace cudaNamespace(namespace_, ss);
+
+  std::string StructName = "globals";
```

Comment:
I think would be weird, because first we fill all the Members, and then we need to create the constructor  GlobalsStruct.addConstructor (out of the loop)